src/main/kotlin/token/ERC20TokenRegistration.kt

```diff
@@ -2,33 +2,32 @@ package token
 
 import com.github.kittinunf.result.Result
 import com.github.kittinunf.result.flatMap
+import com.github.kittinunf.result.map
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.Types
-import jp.co.soramitsu.iroha.Keypair
 import model.IrohaCredential
 import mu.KLogging
-import provider.eth.EthTokenInfo
 import provider.eth.EthTokensProviderImpl
+import sidechain.iroha.consumer.IrohaConsumerImpl
+import sidechain.iroha.util.ModelUtil
 import java.io.BufferedReader
 import java.io.File
 import java.io.FileInputStream
 import java.io.InputStreamReader
 
+/** Information about token - token [name] and [precision] */
+data class EthTokenInfo(val name: String, val precision: Short)
+
 //ERC20 tokens registration class
 class ERC20TokenRegistration(
-    credential: IrohaCredential,
-    private val tokenRegistrationConfig: ERC20TokenRegistrationConfig
+    private val tokenRegistrationConfig: ERC20TokenRegistrationConfig,
+    irohaCredential: IrohaCredential
```

Comment:
what credential is this? can we name it somehow?

Reply:
I added a comment for it.

Reply:
I don't know. For me it's still very confusing. But this is just my opinion.